Tests/Model/AdSet/ParamsTest.php

```diff
@@ -0,0 +1,134 @@
+     * @dataProvider getFieldEnumData
+     *
+     * @params Field $field
+     */
+    public function testAddField(Field $field)
+    {
+        $params = $this->getNewEmptyParams();
+
+        $params->addField($field);
+        $this->assertEquals(
+            ['fields' => $field->getValue()],
+            $params->getParamsArray()
+        );
+        $this->assertEquals('?fields=' . $field->getValue(), $params->getBatchQuery());
+    }
+
+    public function testAddAllFields()
+    {
+        $allField = Field::getValidOptions();
+        $params = $this->getNewEmptyParams();
+
+        $params->addAllFields();
+        $this->assertEquals(
+            ['fields' => implode(', ', $allField)],
+            $params->getParamsArray()
+        );
+    }
+
+    public function testAddFieldsFromString()
+    {
+        $oneParameter = Field::BUDGET_REMAINING;
+        $params = $this->getNewEmptyParams();
+
+        $params->addFieldsFromString($oneParameter);
+        $this->assertEquals(
+            ['fields' => $oneParameter],
+            $params->getParamsArray()
+        );
+
+        $twoParameter = Field::PROMOTED_OBJECT . ', ' . Field::AD_SET_SCHEDULE;
+        $params = $this->getNewEmptyParams();
+
+        $params->addFieldsFromString($twoParameter);
+        $this->assertEquals(
+            ['fields' => $twoParameter],
+            $params->getParamsArray()
+        );
+    }
+
+    /**
+     * @dataProvider getFieldEnumData
+     *
+     * @params Field $field
+     */
+    public function testRemoveField(Field $field)
+    {
+        $params = new Params();
+        $params->addField($field);
+        $this->assertEquals(
+            ['fields' => $field->getValue()],
+            $params->getParamsArray()
+        );
+        $params->removeField($field);
+
+        $this->assertEquals(
+            ['fields' => ''],
+            $params->getParamsArray()
+        );
+
+        $defaultField = Field::get(Field::ACCOUNT_ID);
+        if ($field !== $defaultField) {
+            $parameters = $field->getValue() . ', ' . $defaultField->getValue();
+            $params = new Params();
+            $params->addFieldsFromString($parameters);
+            $this->assertEquals(
+                ['fields' => $parameters],
+                $params->getParamsArray()
+            );
+            $params->removeField($field);
+
+            $this->assertEquals(
+                ['fields' => $defaultField->getValue()],
+                $params->getParamsArray()
+            );
+        }
+    }
+
+    public function testLimit()
+    {
+        $limit = 11;
+        $params = $this->getNewEmptyParams();
+
+        $params->setLimit($limit);
+        $this->assertEquals(['fields' => '', 'limit' => $limit], $params->getParamsArray());
+        $this->assertEquals('?fields=&limit=' . $limit, $params->getBatchQuery());
+    }
+
+
+    /**
+     * @return array
+     */
+    public function getFieldEnumData()
+    {
+        $result = [];
```

Comment:
Call this one data or fieldEnumData so you know what we are talking about. The method suggests that it will return data. As a dataprovider seems to be true. So "result" was unexpected